test/ResultTest.cpp

```diff
@@ -93,107 +93,73 @@ TEST(Result, verifyIdTablesThrowsWhenFullyMaterialized) {
+  auto idTable = makeIdTableFromVector({{1, 6, 0}, {2, 5, 0}, {3, 4, 0}});
 
   for (auto& generator : getAllSubSplits(idTable)) {
-    Result result{std::move(generator), {1, 0}, LocalVocab{}};
+    Result result{std::move(generator), GetParam(), LocalVocab{}};
     AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
         consumeGenerator(result.idTables()),
         ::testing::HasSubstr("compareRowsBySortColumns"),
         ad_utility::Exception);
   }
 
   AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
-      (Result{idTable.clone(), {2, 1}, LocalVocab{}}),
+      (Result{std::move(idTable), GetParam(), LocalVocab{}}),
       ::testing::HasSubstr("compareRowsBySortColumns"), ad_utility::Exception);
-
-  for (auto& generator : getAllSubSplits(idTable)) {
-    Result result{std::move(generator), {2, 1}, LocalVocab{}};
-    AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
-        consumeGenerator(result.idTables()),
-        ::testing::HasSubstr("compareRowsBySortColumns"),
-        ad_utility::Exception);
-  }
 }
 
+INSTANTIATE_TEST_SUITE_P(FailureCases, ResultSortTestF,
+                         testing::Values(CIs{1}, CIs{1, 0}, CIs{2, 1}));
+
 // _____________________________________________________________________________
 TEST(Result,
      verifyAnErrorIsThrownIfSortedByHasHigherIndicesThanTheTableHasColumns) {
   auto idTable = makeIdTableFromVector({{1, 6, 0}, {2, 5, 0}, {3, 4, 0}});
+  using ad_utility::Exception;
+  auto matcher = ::testing::HasSubstr("colIndex < idTable.numColumns()");
 
   AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
-      (Result{idTable.clone(), {3}, LocalVocab{}}),
-      ::testing::HasSubstr("colIndex < idTable.numColumns()"),
-      ad_utility::Exception);
+      (Result{idTable.clone(), {3}, LocalVocab{}}), matcher, Exception);
 
   for (auto& generator : getAllSubSplits(idTable)) {
     Result result{std::move(generator), {3}, LocalVocab{}};
-    AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
-        consumeGenerator(result.idTables()),
-        ::testing::HasSubstr("colIndex < idTable.numColumns()"),
-        ad_utility::Exception);
+    AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(consumeGenerator(result.idTables()),
+                                          matcher, Exception);
   }
 
   AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
-      (Result{idTable.clone(), {2, 1337}, LocalVocab{}}),
-      ::testing::HasSubstr("colIndex < idTable.numColumns()"),
-      ad_utility::Exception);
+      (Result{idTable.clone(), {2, 1337}, LocalVocab{}}), matcher, Exception);
 
   for (auto& generator : getAllSubSplits(idTable)) {
     Result result{std::move(generator), {2, 1337}, LocalVocab{}};
-    AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(
-        consumeGenerator(result.idTables()),
-        ::testing::HasSubstr("colIndex < idTable.numColumns()"),
-        ad_utility::Exception);
+    AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(consumeGenerator(result.idTables()),
+                                          matcher, Exception);
```

Comment:
You could also abstract away the outer structure here (the control loops are basically always the same),
but I don't insist on it.